src/classes/OPP_PrimaryContactRoleMerge.cls

```diff
@@ -0,0 +1,256 @@
+            ocrRecords = new Map<Id, OpportunityContactRole>(loadOpportunityContactRoles(opportunityRecords.values()));
+            opportunityRecordsToUpdate = new Map<Id, Opportunity>();
+            ocrRecordsToUpdate = new Map<Id, OpportunityContactRole>();
+            ocrRecordsToDelete = new Map<Id, OpportunityContactRole>();
+        }
+
+        /*******************************************************************************************************
+        * @description Gets the Opportunities of the duplicate primary OCRs. 
+        * @return List<Opportunity> The Opportunities of the duplicate primary OCRs.
+        ********************************************************************************************************/
+        public List<Opportunity> getOpportunities() {
+            return opportunityRecords.values();
+        }
+
+        /*******************************************************************************************************
+        * @description Sets the OCR in paramater as Primary OCR.
+        * @param OpportunityContactRole the OCR to set as primary.
+        * @return void.
+        ********************************************************************************************************/
+        public void makePrimary(OpportunityContactRole inputOCR) {
+            OpportunityContactRole ocrToUpdate = ocrRecords.get(inputOCR.Id);
+            ocrToUpdate.IsPrimary = true;
+            ocrRecordsToUpdate.put(ocrToUpdate.Id, ocrToUpdate);
+
+            Opportunity opportunityToUpdate = opportunityRecords.get(inputOCR.OpportunityId);
+            if (opportunityToUpdate.Primary_Contact__c != inputOCR.ContactId) {
+                opportunityToUpdate.Primary_Contact__c = inputOCR.ContactId;
+                opportunityRecordsToUpdate.put(opportunityToUpdate.Id, opportunityToUpdate);
+            } 
+        }
+
+        /*******************************************************************************************************
+        * @description Deletes the OCRs in paramater.
+        * @param List<OpportunityContactRole> The OCRs to delete.
+        * @return void.
+        ********************************************************************************************************/
+        public void deleteRecords(List<OpportunityContactRole> ocrsToDelete) {
+            ocrRecordsToDelete.putAll(ocrsToDelete);
+        }
+
+        /*******************************************************************************************************
+        * @description Commits the DMLs (delete/Update OCRs, update Opportunities)
+        * @return void.
+        ********************************************************************************************************/
+        public void processDML() {
+            delete ocrRecordsToDelete.values();
+            update ocrRecordsToUpdate.values();
+            update opportunityRecordsToUpdate.values();
```

Comment:
If we decide to move forward with updating the Opps, do we need exception handling here?

Reply:
Yes, if we update opps we need exception handling with partial success. I really don't think we should be updating opps though.